app/routers/meds.py

```diff
@@ -0,0 +1,424 @@
+    i = 1
+    for time_obj in times:
+        if i <= amount:
+            reminder = datetime.combine(start.date(), time_obj)
+            reminder = adjust_day(reminder, early, time_obj)
+            if reminder > start:
+                if not validate_first_day_reminder(datetime_obj, time_obj,
+                                                   minimum, maximum):
+                    reminder = get_different_time_reminder(
+                        datetime_obj, minimum, early, late)
+                yield reminder
+                datetime_obj = reminder
+                i += 1
+
+
+def reminder_generator(times: List[time], early: time, start: datetime,
+                       day: date, end: datetime) -> Iterator[datetime]:
+    """Generates datetime objects for reminders based on times and date.
+
+    Args:
+        times (list(time)): Reminder times.
+        early (time): Earliest reminder time.
+        start (datetime): First reminder date and time.
+        day (date): Reminders date.
+        end (datetime) - Last reminder date and time.
+
+    Yields:
+        datetime: Reminder datetime object.
+    """
+    for time_obj in times:
+        extra = int(time_obj < early)
+        day_date = start.date() + timedelta(day + extra)
+        reminder = datetime.combine(day_date, time_obj)
+        if reminder <= end:
+            yield reminder
+
+
+def get_reminder_datetimes(form: Dict[str, str]) -> Iterator[datetime]:
+    """Generates datetime object for reminders.
+
+    Args:
+        form (dict(str, str)): Medication form containing all relevant data.
+
+    Yields:
+        datetime: Reminder datetime object.
+    """
+    _, first, _, early, _, _, _, _, start, end = trans_form(form)
+    times = get_reminder_times(form)
+    total_days = (end.date() - start.date()).days + 1
+    for day in range(total_days):
+        if day == 0 and first:
+            yield from get_first_day_reminders(form, times)
+        else:
+            yield from reminder_generator(times, early, start, day, end)
+
+
+def create_events(session: Session, user_id: int,
+                  form: Dict[str, str]) -> None:
+    """Creates reminder events in the DB based on form data.
+
+    Args:
+        session (Session): DB session.
+        user_id (int): ID of user to create events for.
+        form (dict(str, str)): Medication form containing all relevant data.
+    """
+    events = []
+    name, _, _, _, _, _, _, note, _, _ = trans_form(form)
+    title = 'It\'s time to take your meds'
+    if name:
+        title = f'{name.title()} - {title}'
+    datetimes = get_reminder_datetimes(form)
+    for event_time in datetimes:
+        data = {
+            'title': title,
+            'start': event_time,
+            'end': event_time + timedelta(minutes=5),
+            'content': note,
+            'owner_id': user_id,
+        }
+        events.append(create_model(session, Event, **data))
+
+
+@router.get('/')
+@router.post('/')
```

Comment:
I think this is the only function which should stay in `routers/`, the rest should go into `/internals`